Automattic/wp-calypso
# Update/gutenboarding hide next button in intent gathering step

#### Changes proposed in this Pull Request

* Only show the Next button at the top right of the screen if there is a value for Next and a design has been selected.
* Change the label for Next to "Create my site".
* Switch the `setSelectedDesign` behaviour to use the onboard `wordpress/data` store instead of React state hook.
* The "Create my site" button currently just sends the user back to the design step, since we don't (yet) have somewhere for this button to go

#### Before

![image](https://user-images.githubusercontent.com/14988353/71941829-4282fb80-320f-11ea-9fb6-e48aaae86521.png)

#### After

![image](https://user-images.githubusercontent.com/14988353/72124013-579a8e80-33b7-11ea-91ce-1d0571ff68ad.png)

#### Testing instructions



* Go to `http://calypso.localhost:3000/gutenboarding` and in the top right hand corner there shouldn't be a Next button
* Fill out the site topic and site name and click Continue
* On the `/gutenboarding/design` step, after you select a design, you should see the Create my site button at the top right
* Click Back and on the IntentGathering step (the first step), the Next button should not be present at the top right.

Fixes #38698
